### ISSUE TITLE
Windows: ssh coder.workspace not working after running coder config-ssh

### ISSUE THREAD

**Issue by maryna-hankevich** [coder/coder#2853]
Windows 10
Coder version: 0.7.6

Steps:
1) create a workspace
2) run coder config-ssh
3) run ssh coder.workspace

**Actual Result:** 
CreateProcessW failed error:87
posix_spawn: Unknown error
![2022-07-08_18h33_34](https://user-images.githubusercontent.com/35501151/178025591-63771ddb-5762-436a-a0e3-cfcedb1a1903.png)
**Expected Result:** connected successfully

4) correct .ssh file: remove double quotes and double slashes from paths in ProxyCommand
5) run ssh coder.workspace

**Actual=Expected Result:** connected successfully
![2022-07-08_18h34_44](https://user-images.githubusercontent.com/35501151/178026288-d7d4bc40-92fa-499e-8eb8-0b3ef257ce5a.png)

User has to update .ssh file every time after running coder config-ssh to be able to connect via ssh coder.workspace

**Comment by matifali** [comment]
HAs anyone fixed this yet?

**Comment by mafredri** [comment]
@matifali if you have a chance, feel free to test if #3664 fixes the issue for you.

**Comment by matifali** [comment]
Without using fix #3664, I have no issue in Windows 11 build 22622.436, but one of my users using windows 10 has this issue.

**Comment by mafredri** [comment]
Ah, so maybe we should only apply this fix on Win10? Since it’s not required on Win11 or Linux, might be safest that way (to avoid breaking some fringe case).

**Comment by matifali** [comment]
May be we should wait for more people to test this on multiple windows machines before applying anything.

**Comment by maryna-hankevich** [comment]
retested on Windows 10, double slashes were fixed but double quotes still cause the error. SSH connection works only after removing double quotes from config file, probably double quotes are not necessary since the paths do not have spaces
![image](https://user-images.githubusercontent.com/35501151/186896937-882f9402-236c-48b8-9221-8d9fb0f395c4.png)

**Comment by matifali** [comment]
I tested on a different machine having windows 10, and it worked without any fix.

**Comment by mafredri** [comment]
> I tested on a different machine having windows 10, and it worked without any fix.

Thanks for testing that! Just to confirm, does that mean this worked for you?

```
ProxyCommand "D:\\path\\to\\coder.exe" ...
```

Could there be a difference in SSH version used on the machines that are having problems? I guess we need to know more about the environments where it fails. I.e. how was SSH installed and what version.

**Comment by matifali** [comment]
> 

Yes, this worked. And ssh version is the pre-installed version on Windows 10 and 11. ssh version is, 
```powershell
OpenSSH_for_Windows_8.1p1, LibreSSL 3.0.2
```

**Comment by maryna-hankevich** [comment]
@mafredri I retested on Coder v0.8.8+b9b9c2f and it seems to be working fine with double backslashes, but it doesn't work with double quotes, so yes, it looks like double quotes are the only problem
![image](https://user-images.githubusercontent.com/35501151/187196789-ea7afec8-74ce-4a10-a5d5-f5a04d4ce24b.png)

**Comment by mafredri** [comment]
This issue should be fixed by #3755, once merged. However, it will only work when there are no spaces in the path to the Coder binary (or config folder).

If spaces are needed on Windows 10, it should be possible by installing an updated version of OpenSSH from https://github.com/PowerShell/Win32-OpenSSH/releases.
